src/software/replay/replay_reader.cpp

```diff
@@ -0,0 +1,108 @@
+#include "replay_reader.h"
+#include <fstream>
+#include <google/protobuf/util/delimited_message_util.h>
+
+// unfortunately protobuf forces me to write caveman code
+extern "C" {
+#include <unistd.h>
+#include <fcntl.h>
+}
+
+namespace fs = std::experimental::filesystem;
+
+namespace {
+    TbotsReplay readDelimitedReplayProtobufFile(const fs::path& file_path)
+    {
+        // imagine having to write caveman code to read protobuf msgs from files
+        TbotsReplay msg;
+        int fd = open(file_path.c_str(), O_RDONLY);
```

Comment:
should have a try-catch or some check around `open`

Reply:
added